### PR TITLE
Add Soil Properties

### PR DESCRIPTION
**Pull request by mpraprost**
Pull request overview
---------------------

Adds soil conductivity as sampled field in baseline. Change building location measure adds soil conductivity and undisturbed ground temperature as additional properties to the building. These are not used unless the geothermal heat pump upgrade measures are run. Does not affect baseline, no change documentation required. 

 - Fixes #ISSUENUMBERHERE (IF RELEVANT)

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes singularity image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [X] Workflow Measures
 - [ ] Upgrade Measures
 - [ ] Reporting Measures
 - [ ] Postprocessing

Author pull request checklist:

 - [X] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] Added tests for new measures
 - [ ] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [X] Both `options_lookup.tsv` files updated
 - [X] 10k+ test run
 - [ ] Change documentation written
 - [ ] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [x] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: sampling/tsvs/tsvs-v23.zip
Comment (mdahlhausen) [inline sampling/tsvs/tsvs-v23.zip:1]: This file is causing errors. I can unzip v22, but not v23.

File: sampling/tsv_resampling.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def run_sobol_sampling(self, n_datapoints, tmp_csv_path, csv_path):
                 'year_bin_of_last_hvac'
             ],
             [
-                'cook'
+                'cook', 'ground_thermal_conductivity'
```
Comment (mdahlhausen) [inline sampling/tsv_resampling.py]: can this variable be placed with other location variables (climate_zone, region, state, county, etc.? same for the other sampling file.

File: resources/measures/ChangeBuildingLocation/measure.rb [hunk 2]
```diff
@@ -358,6 +364,62 @@ def run(model, runner, user_arguments)
       runner.registerInfo("Setting Climate Zone to #{climateZones.getClimateZones('ASHRAE').first.value}")
     end
 
+    # set soil properties as building additional properties for ground source heat pump modeling
+    soil_conductivity = runner.getDoubleArgumentValue('soil_conductivity', user_arguments)
+
+    # get climate zone and set undisturbed ground temp (used for ground source heat pump modeling)
+    climate_zone = runner.getStringArgumentValue('climate_zone', user_arguments)
+    puts "climate zone = #{climate_zone}"
```
Comment (mdahlhausen) [inline resources/measures/ChangeBuildingLocation/measure.rb]: no puts statements in measures

File: resources/measures/ChangeBuildingLocation/measure.rb [hunk 3]
```diff
@@ -358,6 +364,62 @@ def run(model, runner, user_arguments)
...
+
+    undisturbed_ground_temps = {
```
Comment (mdahlhausen) [inline resources/measures/ChangeBuildingLocation/measure.rb:369]: organize by institution. Ordering by ground temp is confusing.
and check rubocop - the format is off.

File: resources/measures/ChangeBuildingLocation/measure.rb [hunk 4]
```diff
@@ -358,6 +364,62 @@ def run(model, runner, user_arguments)
...
+
+    # Add the values as additional properties to the building
+    building = model.getBuilding
+    building.additionalProperties.setFeature('Soil Conductivity', soil_conductivity)
+    building.additionalProperties.setFeature('Undisturbed Ground Temperature', undisturbed_ground_temp)
+
+    # Report that the measure was successful
+    runner.registerInfo("Added soil conductivity #{soil_conductivity} Btu/hr-ft-F and undisturbed ground temperature #{undisturbed_ground_temp} degC as additional properties to the building.")
+
+    return true
```
Comment (mdahlhausen) [inline resources/measures/ChangeBuildingLocation/measure.rb]: adding return true here misses the final register at end of measure. remove.

**Review by mdahlhausen — CHANGES_REQUESTED** [review]
See comments above. The measure needs changes in the measure.rb, and the tsv v23 zip needs to be regenerated.

Also, please add tests to the changebuildinglocation measure tests for soil conductivity.

File: resources/measures/ChangeBuildingLocation/measure.rb [hunk 5]
```diff
@@ -146,6 +146,12 @@ def arguments(model)
     grid_region.setDefaultValue('RMPAc')
     args << grid_region
 
+    # make argument for soil conductivity (used for ground source heat pump modeling)
+    soil_conductivity = OpenStudio::Ruleset::OSArgument::makeDoubleArgument('soil_conductivity', true)
+    soil_conductivity.setDisplayName('Soil Conductivity')
+    soil_conductivity.setDefaultValue(1.5) # Default value, change as needed
```
Comment (mdahlhausen) [inline resources/measures/ChangeBuildingLocation/measure.rb]: comment is redundant; delete

File: resources/measures/ChangeBuildingLocation/measure.xml
Comment (mdahlhausen) [inline resources/measures/ChangeBuildingLocation/measure.xml:1]: regenerate .xml after changes

**Comment by mpraprost** [comment]
@mdahlhausen please review the changes. the soil properties measure test works but the rest are broken. those may need to get fixed within another PR

**Comment by mdahlhausen** [comment]
@mpraprost I made a few changes to the measure so the tests would run. Now there is a new error in some of the tests where the undisturbed ground temperature lookup is failing to find a match to the climate zone. You'll need to add more error checking, and will need to lookup the climate zone from the model instead of using the measure argument which has 'lookup from stat file' as the default.

**Review by mdahlhausen — APPROVED** [review]

**Pull request merged**